### PR TITLE
[Merged by Bors] - Report received rewards

### LINKED ISSUE #3935: GlobalStateStream doesn't send any rewards
I'm using `GlobalStateStream` with `GlobalStateDataFlag_GLOBAL_STATE_DATA_FLAG_REWARD` flag. I run it for about 40 minutes, and every time reward field is empty.

From what I have been told, current implementation does not stream rewards, the only way to get them is by using `AccountDataQuery`

From explorer perspective, this won't fix issue. I need some way of streaming to get rewards on the fly without looping through every account in the net to check rewards.

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
Closes #3935

## Changes
Report rewards event every time a reward was saved in the DB.

### CODE REVIEW

File: genvm/vm.go [hunk 1]
```diff
@@ -231,6 +231,12 @@ func (v *VM) Apply(lctx ApplyContext, txs []types.Transaction, blockRewards []ty
 		if err := rewards.Add(tx, &reward); err != nil {
 			return nil, nil, fmt.Errorf("%w: %s", core.ErrInternal, err.Error())
 		}
+		events.ReportRewardReceived(events.Reward{
+			Layer:       reward.Layer,
+			Total:       reward.TotalReward,
+			LayerReward: reward.LayerReward,
+			Coinbase:    reward.Coinbase,
+		})
```
Comment (dshulyak) [inline genvm/vm.go]: have you checked if rewards are read from the database when event is received? this is how majority of streams were "designed"
the problem is if event is received before tx is committed, rewards won't be available in the database

Comment (poszu) [reply genvm/vm.go]: I changed it to report rewards events **after** transactions are saved in DB.

**Review by dshulyak — APPROVED** [review]

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**